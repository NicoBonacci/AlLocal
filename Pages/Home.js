import React, { useState, useEffect } from 'react';
import { StyleSheet, TextInput, View, Image, Text, TouchableOpacity } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import * as Location from "expo-location";
import { LogBox } from 'react-native';

import { firebase } from '../react-native-firebase/config'


import logo from '../assets/AllLocal_logo.png';

import locationIcon from '../assets/locationIcon.png';

export default function App({ navigation }) {
    //var usata per salvare la posizione ottenuta dal gps del cell
    const [location, setLocation] = useState(null);

    //var che serve per fare il render solo una volta quando ho i dati del gps altrimenti entra in loop perchï¿½ continua a salvare i dati del gps in location
    const [locUnavailable, setLocUnavailable] = useState(0);




    //var in cui ho la lat long della posizione della mappa
    const [mapRegion, setmapRegion] = useState({
        latitude: 45.473483092360105,
        longitude: 9.187337635878599,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421,
    });
    //inizialmente false, quando clicco su un azienda viene mostrato in basso i dettagli
    const [showDetails, setShowDetails] = useState(false);
    const [companyName, setCompanyName] = useState('');
    const [urlOfImage, setUrlImage] = useState('');
    const [idAzienda, setIdAzienda] = useState('');
    const [descAzienda, setDescAzienda] = useState('');


    const [Allmarkets, setAllMarkets] = useState([]);
    //Scarica i dati dal db una volta sola
    const [downloadMarket, setDownloadMarket] = useState(false);

    //ignora il problema AsyncStorage has been extracted from react-native core and will be removed in a future release. It can now be installed and imported from '@react-native-async-storage/async-storage' instead of 'react-native'. See https://github.com/react-native-async-storage/async-storage
    LogBox.ignoreLogs(['AsyncStorage has been extracted from react-native core and will be removed in a future release.']);

    //ignora il problema each child in a list should... per allMarket.maps quando vado a disegnare i marker in modo iterativo sulla mappa non hanno una key specifica
    LogBox.ignoreLogs(['Each child in a list should have a unique "key" prop.']);

    //controlla se ho l'accesso per ricevere i dati dal gps
    useEffect(() => {
        (async () => {
            let { status } = await Location.requestForegroundPermissionsAsync();
            if (status !== 'granted') {
                return
            }

            let location = await Location.getCurrentPositionAsync({});
            setLocation(location);
        })();
    }, []);

    //imposta la mappa centrata sulla posizione del gps
    if (location && locUnavailable == 0) {
        setmapRegion({
            latitude: location.coords.latitude,
            longitude: location.coords.longitude,
            latitudeDelta: 0.0922,
            longitudeDelta: 0.0421
        })
        setLocUnavailable(1)
    }

    var requestOptions = {
        method: 'GET',
    };

    useEffect(() => {
        if (downloadMarket == false) {
            firebase.firestore().collection('users')
                .get()
                .then(snapshot => {
                    const market = [];
                    snapshot
                        .docs
                        .forEach(doc => {
                            if (doc._delegate._document.data.value.mapValue.fields.isCompany.stringValue == 'Yes') {
                                market.push({
                                    id: doc._delegate._document.data.value.mapValue.fields.id.stringValue,
                                    fullName: doc._delegate._document.data.value.mapValue.fields.fullName.stringValue,
                                    coordination: {
                                        latitude: doc._delegate._document.data.value.mapValue.fields.addreLat.doubleValue,
                                        longitude: doc._delegate._document.data.value.mapValue.fields.addreLong.doubleValue
                                    },
                                    urlPhoto: doc._delegate._document.data.value.mapValue.fields.responseImage.stringValue,
                                    desc: doc._delegate._document.data.value.mapValue.fields.companyDescription.stringValue,
                                });

                            }
                        });
                    setAllMarkets(market);
                });
            setDownloadMarket(true);
        }
    });
    return (
        <View style={styles.main}>
            <View style={styles.up}>
                <View style={styles.up_logo}>
                    <View style={styles.up_logo_enter}>
                        <Image style={{ width: '100%', height: '100%' }}
                            source={logo}
                        />
                    </View>
                </View>
                <View style={styles.up_search}>
                    <View style={styles.up_search_enter}>
                        {//ottiene la posizione della citta cercata nella barra di ricerca
                        }
                        <TextInput
                            style={styles.searchBar}
                            placeholder={'Search'}
                            placeholderTextColor={'#666'}
                            onSubmitEditing={(event) => {
                                fetch("https://api.geoapify.com/v1/geocode/search?text=" + event.nativeEvent.text + "&apiKey=e5e4640f03c147c58545023a81e1b8e1", requestOptions)
                                    .then(response => response.json())
                                    .then(result => {
                                        setmapRegion({
                                            latitude: result.features[0].geometry.coordinates[1],
                                            longitude: result.features[0].geometry.coordinates[0],
                                            latitudeDelta: 0.0922,
                                            longitudeDelta: 0.0421
                                        })

                                    })
                                    .catch(error => console.log('error', error));
                            }

                            } />
                    </View>
                </View>

            </ View>
            <View style={styles.center} >


                <MapView loadingEnabled={true}
                    style={styles.mapViewStyle}
                    region={mapRegion}
                    onRegionChangeComplete={(region) => {
                        setmapRegion({
                            latitude: region.latitude,
                            longitude: region.longitude,
                            latitudeDelta: region.latitudeDelta,
                            longitudeDelta: region.longitudeDelta
                        })
                    }
                    }>
                    {
                        //carica la posizione del gsp se ho i valori, se ho i val me lo dice locUnavalable
                    }
                    {locUnavailable ?
                        <Marker coordinate={{
                            latitude: location.coords.latitude,
                            longitude: location.coords.longitude,
                            latitudeDelta: 0.0922,
                            longitudeDelta: 0.0421
                        }} title='Your position'
                            onPress={() => { }}
                        ><Image source={locationIcon}
                            style={{ height: 20, width: 20 }} /></Marker> : null
                    }


                    {Allmarkets ? Allmarkets.map((market) => (

                        <Marker coordinate={market.coordination}
                            title={market.fullName}
                            onPress={() => {
                                setShowDetails(true)
                                setCompanyName(market.fullName)
                                setUrlImage(market.urlPhoto)
                                setIdAzienda(market.id)
                                setDescAzienda(market.desc)
                            }
                            }
                        />

                    )) : null
                    }
                    {showDetails ?
                        <View style={styles.down}>
                        <TouchableOpacity style={styles.down_company} onPress={() => navigation.navigate('Azienda', { desc: descAzienda, id: idAzienda, name: companyName })}>
                            <Text style={styles.textCompany}>{companyName}</Text>
                            <Image style={styles.picCompanyDimension} source={{ uri: urlOfImage }} />
                        </TouchableOpacity>
                        </View>
                        : null}

                </MapView>

            </View>

        </View>
    );

}

const styles = StyleSheet.create({

    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },

    main: {
        flex: 1,
        width: '100%',
        height: '100%',
        alignItems: 'flex-start',
        padding: 10,
    },
    up: {
        flexDirection: 'row',
        width: '100%',
        height: '15%',
    },
    center: {
        width: '100%',
        height: '50%',
        marginHorizontal: 0,
        flexGrow: 1,

    },
    down: {
        width: '100%',
        height: '29%',
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    up_logo: {
        width: '20%',
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',
    },
    up_search: {
        width: '80%',
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',

    },
    down_company: {
        width: '92%',
        height: '90%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#EAEAEA',
    },
    up_logo_enter: {

        width: '100%',
        height: '70%',

    },
    up_search_enter: {

        width: '100%',
        height: '40%',
    },
    searchBar: {
        borderRadius: 10,
        margin: 10,
        color: '#000',
        borderColor: '#666',
        backgroundColor: '#FFF',
        borderWidth: 1,
        height: 45,
        paddingHorizontal: 10,
        fontSize: 18,
    },
    mapViewStyle: {
        alignSelf: 'stretch',
        height: '100%'
    },
    textCompany: {
        fontSize: 25,

    },
    picCompanyDimension: {
        marginTop: 15,
        width: '50%',
        height: '50%',
    }
});

