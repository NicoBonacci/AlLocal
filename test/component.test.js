import * as React from "react";

import Home from "../Pages/home";
import CambioPassword from "../Pages/cambioPassword";
import EditBioAzienda from "../Pages/editBioAzienda";
import EditProdotto from "../Pages/editProdotto";
import PicUpdate from "../Pages/picUpdate";


import renderer from "react-test-renderer";

it(`renders correctly`, () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
    const tree = renderer.create(<CambioPassword />);
    expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
    const tree = renderer.create(<EditBioAzienda />);
    expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
    const tree = renderer.create(<EditProdotto />);
    expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
    const tree = renderer.create(<PicUpdate />);
    expect(tree).toMatchSnapshot();
});

