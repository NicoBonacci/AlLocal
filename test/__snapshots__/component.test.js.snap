// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#C6EBBE",
      "flex": 1,
      "height": "100%",
      "padding": "3%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "15%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "height": "40%",
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onSubmitEditing={[Function]}
          placeholder="Search"
          placeholderTextColor="#666"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#666",
              "borderRadius": 10,
              "borderWidth": 1,
              "color": "#000",
              "fontSize": 18,
              "height": 45,
              "margin": 10,
              "paddingHorizontal": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexGrow": 1,
        "height": "85%",
        "marginHorizontal": 0,
        "width": "100%",
      }
    }
  >
    <AIRMap
      handlePanDrag={false}
      initialRegion={null}
      loadingEnabled={true}
      onChange={[Function]}
      onLayout={[Function]}
      onMapReady={[Function]}
      onMarkerPress={[Function]}
      onRegionChangeComplete={[Function]}
      region={
        Object {
          "latitude": 45.473483092360105,
          "latitudeDelta": 0.0922,
          "longitude": 9.187337635878599,
          "longitudeDelta": 0.0421,
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "height": "100%",
          "padding": "10%",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#C6EBBE",
      "borderBottomWidth": 1,
      "borderColor": "#000",
      "flex": 2,
      "padding": "2%",
    }
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Insert new password"
      placeholderTextColor="#aaaaaa"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 48,
          "marginBottom": 10,
          "marginLeft": 30,
          "marginRight": 30,
          "marginTop": 10,
          "overflow": "hidden",
          "paddingLeft": 16,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm Password"
      placeholderTextColor="#aaaaaa"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "height": 48,
          "marginBottom": 10,
          "marginLeft": 30,
          "marginRight": 30,
          "marginTop": 10,
          "overflow": "hidden",
          "paddingLeft": 16,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D95D39",
          "borderRadius": 5,
          "height": 48,
          "justifyContent": "center",
          "marginLeft": 30,
          "marginRight": 30,
          "marginTop": 20,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Cambia password
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "backgroundColor": "#C6EBBE",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          multiline={true}
          numberOfLines={5}
          onChangeText={[Function]}
          placeholder="insert here new biography"
          placeholderTextColor="#aaaaaa"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "height": 48,
              "marginBottom": 10,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 10,
              "overflow": "hidden",
              "paddingLeft": 16,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D95D39",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "opacity": 1,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                 PICK PICTURE 
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D95D39",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "opacity": 1,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                 PICK CAMERA 
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": null,
                }
              }
              style={
                Object {
                  "backgroundColor": "black",
                  "height": 200,
                  "width": 200,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D95D39",
              "flex": 1,
              "height": 50,
              "justifyContent": "center",
              "marginLeft": "13%",
              "marginTop": "13%",
              "opacity": 1,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            SAVE
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#C6EBBE",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Insert new name"
          placeholderTextColor="#aaaaaa"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "height": 48,
              "marginBottom": 10,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 10,
              "overflow": "hidden",
              "paddingLeft": 16,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          multiline={true}
          numberOfLines={5}
          onChangeText={[Function]}
          placeholder="Insert new description"
          placeholderTextColor="#aaaaaa"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "height": 48,
              "marginBottom": 10,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 10,
              "overflow": "hidden",
              "paddingLeft": 16,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Insert new price"
          placeholderTextColor="#aaaaaa"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "height": 48,
              "marginBottom": 10,
              "marginLeft": 30,
              "marginRight": 30,
              "marginTop": 10,
              "overflow": "hidden",
              "paddingLeft": 16,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D95D39",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "opacity": 1,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                 PICK PICTURE 
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#D95D39",
                  "flex": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "margin": 8,
                  "opacity": 1,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                 PICK CAMERA 
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "width": "100%",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": null,
                }
              }
              style={
                Object {
                  "backgroundColor": "black",
                  "height": 200,
                  "width": 200,
                }
              }
            />
          </View>
        </View>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D95D39",
              "flex": 1,
              "height": 50,
              "justifyContent": "center",
              "marginLeft": "13%",
              "marginTop": "13%",
              "opacity": 1,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            SAVE CHANGE
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders correctly 5`] = `<View />`;
